packages/ui/components/core/test/SlotMixin.test.js

```diff
@@ -195,102 +142,361 @@ describe('SlotMixin', () => {
 
-        render() {
-          return html`<slot name="template"></slot>`;
-        }
-      },
-    );
-    const el = await fixture(`<${tag}></${tag}>`);
-    const slot = /** @type HTMLSpanElement */ (
-      Array.from(el.children).find(elm => elm.slot === 'template')
-    );
-    expect(slot.slot).to.equal('template');
-    expect(slot.tagName).to.equal('SPAN');
-  });
+          constructor() {
+            super();
+            this.currentValue = 0;
+          }
 
-  it('supports scoped elements when polyfill loaded', async () => {
-    const outputObj = mockScopedRegistry();
+          get slots() {
+            return {
+              ...super.slots,
+              template: () => ({ template: html`<span>${this.currentValue}</span> ` }),
+            };
+          }
 
-    class ScopedEl extends LitElement {}
+          render() {
+            return html`<slot name="template"></slot>`;
+          }
 
-    const tagName = defineCE(
-      class extends ScopedElementsMixin(SlotMixin(LitElement)) {
-        static get scopedElements() {
-          return {
+          get _templateNode() {
+            return /** @type HTMLSpanElement */ (
+              Array.from(this.children).find(elm => elm.slot === 'template')
+            );
+          }
+        },
+      );
+      const el = /** @type {* & SlotHost} */ (await fixture(`<${tag}></${tag}>`));
+
+      expect(el._templateNode.slot).to.equal('template');
+      expect(el._templateNode.textContent?.trim()).to.equal('0');
+
+      el.currentValue = 1;
+      await el.updateComplete;
+
+      expect(el._templateNode.textContent?.trim()).to.equal('1');
+    });
+
+    it('keeps focus after rerender', async () => {
+      const tag = defineCE(
+        // @ts-expect-error
+        class extends SlotMixin(LitElement) {
+          static properties = { currentValue: Number };
+
+          constructor() {
+            super();
+            this.currentValue = 0;
+          }
+
+          get slots() {
+            return {
+              ...super.slots,
+              'focusable-node': () => ({
+                template: html`<input /> `,
+                // afterRender: () => {
+                //   varThatProvesAfterRenderIsCalled = 'called';
+                // },
+                createRenderWrapper: true,
+              }),
+            };
+          }
+
+          render() {
+            return html`<slot name="focusable-node"></slot>`;
+          }
+
+          get _focusableNode() {
+            return /** @type HTMLSpanElement */ (
+              Array.from(this.children).find(elm => elm.slot === 'focusable-node')
+                ?.firstElementChild
+            );
+          }
+        },
+      );
+      const el = /** @type {* & SlotHost} */ (await fixture(`<${tag}></${tag}>`));
+
+      el._focusableNode.focus();
+      expect(document.activeElement).to.equal(el._focusableNode);
+
+      // el.currentValue = 1;
+      // await el.updateComplete;
```

Comment:
this is not executed? so what is it testing then? 🤔

Reply:
Valid point. These are some leftovers of a previous approach. Will look into it

Reply:
Should be fixed 👍